src/server/main.py

```diff
@@ -1,30 +1,345 @@
+        try:
+            data: dict[str, Any] = jwt.decode(token, app.config["SECRET_KEY"], algorithms=["HS256"])
+        except jwt.InvalidTokenError as e:
+            return make_response(
+                json.dumps({"message": f"Token is invalid: {e}"}),
+                401,
+                {"WWW-Authenticate": 'Basic realm ="Token is invalid!"'},
+            )
+
+        try:
+            connection: Connection = getDatabaseConnection()
+            sql: str = "SELECT uuid, email FROM users WHERE uuid=?"
+            cursor: Cursor = connection.cursor()
+            cursor.execute(sql, (data["uuid"],))
+            user_information: list[tuple[str, str]] = cursor.fetchall()
+            connection.close()
+        except SqlError as e:
+            return make_response(json.dumps({"message": f"Database connection error: {e}"}), 500)
+
+        unique_id: str = ""
+
+        if not user_information:
+            return make_response(
+                json.dumps({"message": "User does not exist"}),
+                401,
+                {"WWW-Authenticate": 'Basic realm ="User does not exist!"'},
+            )
+
+        # Retrieve user uuid
+        unique_id = user_information[0][0]
+
+        # Returns the current logged-in users context to the routes
+        return fun(unique_id, *args, **kwargs)
+
+    return decorated
+
+
+def revoke_token(token: str) -> None:
+    """Revoking activated token."""
+    decoded_token: dict[str, Any] = jwt.decode(
+        token, app.config["SECRET_KEY"], algorithms=["HS256"]
+    )
+    expiry: datetime = datetime.fromtimestamp(decoded_token["exp"])
+
+    # Saving revoked token into the database
+    try:
+        connection: Connection = getDatabaseConnection()
+        connection.execute(
+            "INSERT INTO revoked_tokens (token, expiry) VALUES (?, ?)",
+            (
+                token,
+                expiry,
+            ),
+        )
+        connection.commit()
+        connection.close()
+    except SqlError as e:
+        raise SqlError(f"Database connection error: {str(e)}") from e
+
+
+def is_token_revoked(token: str) -> bool:
+    """Check if token is revoked."""
+    try:
+        connection: Connection = getDatabaseConnection()
+        result: Cursor = connection.execute(
+            "SELECT token FROM revoked_tokens WHERE token=? AND expiry > ?",
+            (
+                token,
+                datetime.utcnow(),
+            ),
+        )
+        is_revoked: bool = result.fetchone() is not None
+        connection.close()
+    except SqlError:
+        return True
+
+    return is_revoked
+
+
+@app.route("/register", methods=["POST"])
+def register() -> Response:
+    """Registration endpoint."""
+    new_user: dict[str, str] = request.json
+
+    unique_id: UUID = uuid.uuid4()
+    try:
+        email: str = new_user["email"]
+        password: str = new_user["password"]
+        repeated_password: str = new_user["repeatedPassword"]
```

Comment:
I am not sure if checking for repeated password should appear on the backend at all. This is just a convenience check on frontend to ensure that the user didn't misstype something. Do we even receive this key from frontend?

Reply:
As I mentioned before, I discussed this with the frontend leader and we're receiving this key from the frontend. After reconsidering, I agree that it may not be a good idea to perform this type of check on the backend. If we kept checking password correctness on the backed side, we should also check correctness of the other paramters to ensure consistency. Therefore I removed the password check from the backend code.